src/symmetric.jl

```diff
@@ -461,9 +461,34 @@ issymmetric(A::Hermitian{<:Real}) = true
 issymmetric(A::Hermitian{<:Complex}) = isreal(A)
 issymmetric(A::Symmetric) = true
 
-# check if the symmetry is known from the type
-_issymmetric(::Union{SymSymTri, Hermitian{<:Real}}) = true
-_issymmetric(::Any) = false
+"""
+    issymmetrictype(T::Type)
+
+Return whether every instance `x` of the type `T` satisfies `issymmetric(x) == tue`,
+that is, the fact that the instance is symmetric is known from its type.
+
+!!! note
+    An instance `x::T` may still be symmetric when `issymmetrictype(T)` returns `false`.
+"""
+issymmetrictype(::Type) = false
+issymmetrictype(::Type{<:Union{Symmetric,Hermitian{<:Real}}}) = true
+issymmetrictype(::Type{<:Real}) = true
+issymmetrictype(::Type{<:AbstractFloat}) = false
```

Comment:
Shouldn't it be
```
issymmetrictype(::Type{<:Number}) = true
```
?

Reply:
The exception is `NaN`, which isn't equal to itself.